download-chromedriver.js

```
@@ -1,41 +1,28 @@
-
-function download (version) {
-  return downloadArtifact({
-    version,
-    artifactName: 'chromedriver',
-    platform: process.env.npm_config_platform,
-    arch: process.env.npm_config_arch,
-    rejectUnauthorized: process.env.npm_config_strict_ssl === 'true',
-    quiet: ['info', 'verbose', 'silly', 'http'].indexOf(process.env.npm_config_loglevel) === -1
-  })
-}
+const os = require('os');
+const { DownloaderHelper } = require('node-downloader-helper')
 
 async function attemptDownload (version) {
-  try {
-    const targetFolder = path.join(__dirname, 'bin')
-    const zipPath = await download(version)
-    await extractZip(zipPath, { dir: targetFolder })
-    const platform = process.env.npm_config_platform || process.platform
-    if (platform !== 'win32') {
-      await fs.chmod(path.join(targetFolder, 'chromedriver'), 0o755)
-    }
-  } catch (err) {
-    // attempt to fall back to semver minor
-    const parts = version.split('.')
-    const baseVersion = `${parts[0]}.${parts[1]}.0`
 
-    // don't recurse infinitely
-    if (baseVersion === version) {
-      throw err
-    } else {
-      await attemptDownload(baseVersion)
+    try {
+      const targetFolder = path.join(os.homedir(), 'Library/Caches/electron')
+      if (fs.existsSync(targetFolder + '/chromedriver-v10.1.5-darwin-x64.zip')) {
+        console.log('chromedriver already downloaded!')
+      } else {
+        const dl = new DownloaderHelper
+        ("https://github.com/electron/electron/releases/download/v10.1.5/chromedriver-v10.1.5-darwin-x64.zip", targetFolder)
+        dl.on('end', () => console.log('Download Completed'))
+        dl.start();
+        const platform = process.env.npm_config_platform || process.platform
+        if (platform !== 'win32') {
+        await fs.chmod(path.join(targetFolder, 'chromedriver-v10.1.5-darwin-x64.zip'), 0o755, () => { console.log('chmod done')})
+        }
+      }
+    } catch(err) {
+      console.error(err)
```

this isn't correct, this will swallow the error and return exit code 0 when there is a failure. we need it to let the error happen, so it doesn't pretend everything is ok.